pkg/storage/etcd3/watcher.go

```diff
@@ -180,6 +180,9 @@ func (wc *watchChan) startWatching() {
 			wc.sendEvent(parseEvent(e))
 		}
 	}
+	wc.cancel()
```

Comment:
ok. now i see we also do w.cancel at line 120. i think the logic is not clear. probably we should always cancel the watcher in run func?